pyproject.toml

```diff
@@ -152,6 +152,10 @@ Metrics = [
     '.spin/cmds.py:asv'
 ]
 
+[tool.black]
+target-version = ['py39', 'py310', 'py311', 'py312']
+skip-string-normalization = 1
```

Comment:
This tells black to leave quotes as is.

Reply:
Is everyone ok with 90c line lengths?

Reply:
I much prefer it.

Reply:
I prefer having the 8 or 9 extra characters per line.

Reply:
In case anyone wants to compare, I made a [branch](https://github.com/jarrodmillman/scikit-image/tree/black2) with the line length set to `79` instead of the default `88`.
﻿
Using the line length set to `79` produces an extra 5690.
﻿
In particular, with `88` characters per line the PR has
﻿
`448 changed files with 20,741 additions and 13,566 deletions.`
﻿
and  with `79` characters per line the PR has
﻿
`456 changed files with 26,055 additions and 13,190 deletions.`
﻿
Scanning the files, I prefer the text with `88` characters per line still; but like @stefanv, I don't care strongly about it.

Reply:
As someone who uses 80c terminal windows because they thought PEP8 was sacrosanct, I am sad. But holiness is dead, it seems. 😂

Reply:
Have you considered an 18 inch gaming laptop?

Reply:
Battery life was too shit. 😜

Reply:
For the record, my preference also defaults to 88c, though I don't have experience with black & 79c. Just with manual 79c which makes me feel like I subconsciously start avoiding long variable names (might be total imagination though). :sweat_smile: 

Concerning argument about fitting X columns, it seems to me that 79c is one variable of many. E.g. DPI, font size, monitor width, eye sight, ... Not sure how generalizable that is but I get the argument. I'm quite annoyed if something is messing with my workflow.

Reply:
![image](https://github.com/scikit-image/scikit-image/assets/45071/6aebf234-bfb6-4e27-83b3-2acece673a2f)

I've been meaning to ask whether we can also support 40 character line lengths for my slightly atypical setup.

Reply:
Personally, I'm fine with 88c, especially in light of the stats @jarrodmillman shared in https://github.com/scikit-image/scikit-image/pull/7197#discussion_r1357240465 (thanks for this, btw). It's just that I'll probably keep wrapping (most) lines I write at 79c.

Reply:
https://github.com/astral-sh/ruff/issues/8683#issuecomment-1811369165

Reply:
Looks like it was easy enough to fix. 😉

Reply:
I mainly wanted to highlight the colorful term: "80 chars line limit olympics" :)

Reply:
Look if you want to abandon the commitment to excellence and olympic glory and settle for the mediocrity of the "Black 88 char line regional athletics meet", be my guest. 😜

Reply:
It is well known that our eyes, by default, can only take in 50-75 characters without jumping. That is the next axis along which we need to train ourselves to improve. For that purpose, I suggest decreasing font size by 9.1%, while keeping any existing terminal layout.

Reply:
If we fail to retrain our biology, there is still hope: since Black is going to do Whatever It Wants with our code anyway, there is no reason not to use an automated tool to rewrap to 80 characters before editing! We call this Virtualized Wrapping. The robustness of such a roundtripping approach is left as an exercise for the developer.

Reply:
That's actually pretty genius @stefanv! I don't ever have to look at black formatting again if I set up my post-checkout hooks correctly! 😍